### PR TITLE
Make figure tests work with pytest-mpl 0.12

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: ndcube/tests/helpers.py [hunk 1]
```diff
@@ -3,22 +3,74 @@
...
+    sunpy_version = "dev" if "dev" in sunpy.__version__ else sunpy.__version__.replace('.', '')
+    astropy_version = "dev" if "dev" in astropy.__version__ else astropy.__version__.replace('.', '')
+    return f"figure_hashes_mpl_{mpl_version}_ft_{ft2_version}_astropy_{astropy_version}_sunpy_{sunpy_version}.json"
+
+
+def figure_test(test_function):
+    """
+    A decorator for a test that verifies the hash of the current figure or the
+    returned figure, with the name of the test function as the hash identifier
+    in the library. A PNG is also created in the 'result_image' directory,
```
Comment (DanRyanIrish) [inline ndcube/tests/helpers.py:46]: Is the result_image a tmp dir or is it actually in the repo?

File: tox.ini [hunk 2]
```diff
@@ -36,15 +26,35 @@ changedir = .tmp/{envname}
...
     test
 deps =
     pytest-xdist
-    devdeps: git+https://github.com/sunpy/sunpy
     devdeps: git+https://github.com/astropy/astropy
+    devdeps: git+https://github.com/sunpy/sunpy
+
+    # Figure tests need a tightly controlled environment
+    figure: matplotlib==3.3.2
+    figure-!devdeps: astropy==4.1.0
```
Comment (DanRyanIrish) [inline tox.ini:51]: What does ```-!``` mean?

**Review by DanRyanIrish — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (Cadair) [reply tox.ini:51]: Not that factor so `-figure` would match but `-figure-devdeps` would not.

File: ndcube/tests/helpers.py (hunk 1 above)
Comment (Cadair) [reply ndcube/tests/helpers.py:46]: This docstring is well out of date, will nuke.

**Pull request merged**